andela/team-odd-bn-backend
# #169258622 Setup unit testing and test coverage.

#### What does this PR do?
Sets up the testing environment as well as test coverage for the project

#### Description of Task to be completed?
- Install testing framework `mocha`, assertion library ` chai `and ` nyc` for test coverage
- Integrate Coveralls and Code Climate.
- Add test and coverage scripts in package JSON file

#### How should this be manually tested?
1. Clone this repo using the command `git clone https://github.com/andela/team-odd-bn-backend.git`
2. Run the command `npm install` to install all dependencies
3. Checkout to this branch  using the command `git checkout ch-setup-unit-testing-and-coveralls-169258622`
4. run the command `npm run test` to check for the testing environment 
5. run the command `npm run coverage` to check for the test coverage

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169258622](https://www.pivotaltracker.com/story/show/169258622)

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2019-11-07 at 11 31 09" src="https://user-images.githubusercontent.com/45204494/68378587-0a026b00-0155-11ea-913a-da5bf321cc58.png">
